src/providers/foldersCompareProvider.ts

```diff
@@ -19,16 +20,31 @@ export class CompareFoldersProvider implements TreeDataProvider<File> {
   private workspaceRoot: string;
 
   constructor(private onlyInA: ViewOnlyProvider, private onlyInB: ViewOnlyProvider) {
-    this.workspaceRoot = workspace.workspaceFolders ? workspace.workspaceFolders[0].uri.path : '';
+    this.workspaceRoot = workspace.workspaceFolders ? workspace.workspaceFolders[0].uri.fsPath : '';
   }
 
 	chooseFoldersAndCompare = async () => {
-    const diffs = await chooseFoldersAndCompare(await this.getWorkspaceFolder());
-    if (!diffs) {
-      return;
-    }
-    this._diffs = diffs;
-    this.updateUI();
+    await window.withProgress({
```

Comment:
Maybe the progress indicator should be wrap only the comparing operation?

https://github.com/moshfeu/vscode-compare-folders/pull/11/files#diff-c4be055bd42675021809ec3ae550dedeR41

Reply:
NM, I'll merge it now and we can discuss about it later, it's not that important..